src/components/Modal/Modal.tsx

```diff
@@ -114,15 +118,33 @@ export const Modal = forwardRef(
       isBodyPadding,
       headingAlign = 'left',
       footerAlign = 'right',
-      hasEnterKeyDown
+      hasEnterKeyDown,
+      hasAnimation = false
     }: ModalProps,
     ref: Ref<HTMLDivElement>
   ) => {
+    const [isClosing, setClosing] = useState(false)
+
+    const [blockScroll, allowScroll] = useScrollBlock()
+
     const headerStyles = useStyles({
       [styles.modal__header]: true,
-      [styles['modal__header--close-only']]: !i18n.title && !image
+      [styles['modal__header--close-only']]: !i18n.title && !image,
+      [styles['modal__header--animation-open']]: hasAnimation,
```

Comment:
animacja może być tylko jak jest fullscreen, warto by było to dodać i uwzględnić w typach - nie jest fullscreen, nie możesz dodać tego propsa :P Będziesz miał trochę zabawy z tym ale dasz radę 💪

Reply:
jakby były jakieś problemy do uderzaj do mnie

Reply:
Spoko. Spróbuje to tak ogarnąć 💪